### PR TITLE
Feature/survival plot api

### PR DESCRIPTION
**Pull request by andricDu**
`POST http://localhost:8080/api/v1/analysis/survival`

```
Request Body: 
["53cc800f-507b-4c18-9102-68a2bf50cb00", "cc4195c6-0ff0-4c0f-a469-5b79bad67f4d"]
```

`GET http://localhost:8080/api/v1/analysis/survival/<uuid>`

### CODE REVIEW

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+      }
+      return sum;
+    }
+
+  }
+
+  @Value
+  public static class DonorValue {
+
+    private final String id;
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java]: `private final` not needed

File: dcc-portal-api/src/main/sql/schema.sql [hunk 2]
```diff
@@ -64,6 +64,17 @@ GRANT SELECT, INSERT, UPDATE, DELETE ON phenotype_analysis TO dcc;
 
 -------------------------------------------------------------------------------- 
 
+CREATE TABLE IF NOT EXISTS survival_analysis(
```
Comment (btiernay) [inline dcc-portal-api/src/main/sql/schema.sql]: Don't forget to deploy with Bob G.!

File: dcc-portal-api/src/test/sql/schema.sql [hunk 3]
```diff
@@ -45,6 +45,14 @@ CREATE TABLE IF NOT EXISTS phenotype_analysis(
 
 --------------------------------------------------------------------------------
 
+CREATE TABLE IF NOT EXISTS survival_analysis(
```
Comment (btiernay) [inline dcc-portal-api/src/test/sql/schema.sql]: Don't forget to deploy to dev(s)!

**Comment by andricDu** [comment]
Example output from the `GET`:

```
{
  "id": "8359e198-3137-4ec8-adf7-b40f26ea7c97",
  "entitySetIds": [
    "53cc800f-507b-4c18-9102-68a2bf50cb00",
    "cc4195c6-0ff0-4c0f-a469-5b79bad67f4d"
  ],
  "inputCount": 2,
  "version": 2,
  "state": "FINISHED",
  "timestamp": 1467053341924,
  "results": [
    {
      "id": "53cc800f-507b-4c18-9102-68a2bf50cb00",
      "intervals": [
        {
          "start": 0,
          "end": 2,
          "died": 1,
          "donors": [
            {
              "id": "DO221140",
              "status": "deceased",
              "survivalTime": 2
            }
          ],
          "cumulativeSurvival": 1,
          "censured": 0
        },
.
.
.
```

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    }
+    currentInterval.setCumulativeSurvival(cumulativeSurvival);
+
+    return intervals;
+  }
+
+  @Data
+  public static class Interval {
+
+    private int start;
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java]: `final`

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+    private int end;
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java]: `final`

File: dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    private int died;
+    private List<DonorValue> donors = new ArrayList<DonorValue>();
+    private float cumulativeSurvival;
+
+    public Interval(int start, int end) {
```
Comment (btiernay) [inline dcc-portal-api/src/main/java/org/icgc/dcc/portal/analysis/SurvivalAnalyzer.java]: Remove then.

**Pull request merged**